### ISSUE TITLE
[QUESTION] --fp4-format e2m1 & --fp4-recipe nvfp4 does not seem to work.

### ISSUE THREAD

**Issue by deepak-vij** [NVIDIA/Megatron-LM#3423]
**Your question**
I am using Miles framework with Megatron as the backend for VLM supervised fine-tuning (SFT) on B200. I am trying to enable nvfp4 processing for my SFT training run using arguments --fp4-format 'e2m1' --fp4-recipe nvfp --fp4-param-gather. The SFT training runs successfully.  But for some reason, I don't see the performance benefits of enabling nvfp4 datatype at the time of SFT training. I don't see any error/s in the log file. Is there something else I am missing? Also, I don't see any nvfp4 specific kernels being invoked too in the pytorch/profiler Perfetto traces.

**Comment by Phlip79** [comment]
Could you please provide a reproducer? Also, how are you measuring performance?

**Comment by deepak-vij** [comment]
Hi @Phlip79 , thanks for responding. Below are my SFT arguments. As I mentioned that I don't see any trace of nvfp4 specific kernels in my pytorch profiler logs. 

SFT_ARGS=(
   --rollout-function-path miles.rollout.sft_rollout.generate_rollout
   --prompt-data /root/datasets/${DATASET_LOCAL_NAME}/miles_sft_final.jsonl
   --input-key messages
   --rollout-shuffle
   --num-epoch 100
   --rollout-batch-size 96
   --transformer-impl transformer_engine
   --fp4-format 'e2m1'
   --fp4-recipe nvfp4
   --fp4-param-gather
   
   --num-steps-per-rollout 1
   
   --loss-type sft_loss
   --calculate-per-token-loss
   --disable-compute-advantages-and-returns
   --debug-train-only
   --tensor-model-parallel-size 4
)

**Comment by deepak-vij** [comment]
As a first step, it would great to confirm if the nvfp4 aware SFT process is triggered using the Transform_Engine API. What exactly should I look for in the Pytorch Profiler or other logs to ascertain this? Thanks.

**Comment by deepak-vij** [comment]
@Phlip79 , I think I finally found the reason why nvfp4/mxfp8 does not work for miles/megatron SFT environment. The culprit is the docker image "radixark/miles:latest" I have been using. This image does not have latest Megatron code in it.

Now I need to figure out if I need to build the miles/megatron from scratch within the existing container, in case of any dependencies issues. I also see some modification to the Megatron codebase in this repo. Or maybe I can get away with retrofitting the existing container with the latest Megatron codebase that seems to have nvfp4 capability as part of /core/extensions/transformer_engine.py module.

**Comment by Phlip79** [comment]
Glad to hear your issue is resolved.
